qt/widgets/common/src/WorkspaceMultiSelector.cpp

```diff
@@ -80,67 +77,37 @@ void WorkspaceMultiSelector::setupTable() {
 /**
  * De-subscribes this object from the Poco NotificationCentre
  */
-void WorkspaceMultiSelector::disconnectObservers() {
-  if (m_init) {
-    Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_addObserver);
-    Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_remObserver);
-    Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_clearObserver);
-    Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_renameObserver);
-    Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_replaceObserver);
-    m_init = false;
-  }
+void WorkspaceMultiSelector::disconnectObservers() const {
+  Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_addObserver);
+  Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_remObserver);
+  Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_clearObserver);
+  Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_renameObserver);
+  Mantid::API::AnalysisDataService::Instance().notificationCenter.removeObserver(m_replaceObserver);
 }
 
 /**
  * Subscribes this object to the Poco NotificationCentre
  */
-void WorkspaceMultiSelector::connectObservers() {
+void WorkspaceMultiSelector::connectObservers() const {
   Mantid::API::AnalysisDataServiceImpl &ads = Mantid::API::AnalysisDataService::Instance();
   ads.notificationCenter.addObserver(m_addObserver);
   ads.notificationCenter.addObserver(m_remObserver);
   ads.notificationCenter.addObserver(m_renameObserver);
   ads.notificationCenter.addObserver(m_clearObserver);
   ads.notificationCenter.addObserver(m_replaceObserver);
-  refresh();
-  m_init = true;
-}
-
-QStringList WorkspaceMultiSelector::getWorkspaceTypes() const { return m_workspaceTypes; }
-
-void WorkspaceMultiSelector::setWorkspaceTypes(const QStringList &types) {
-  if (types != m_workspaceTypes) {
-    m_workspaceTypes = types;
-    if (m_init) {
-      refresh();
-    }
-  }
-}
-
-bool WorkspaceMultiSelector::showWorkspaceGroups() const { return m_showGroups; }
-
-void WorkspaceMultiSelector::showWorkspaceGroups(bool show) {
-  if (show != m_showGroups) {
-    m_showGroups = show;
-    if (m_init) {
-      refresh();
-    }
-  }
 }
 
 bool WorkspaceMultiSelector::isValid() const {
   const QTableWidgetItem *item = currentItem();
   return (item != nullptr);
 }
 
-QStringList WorkspaceMultiSelector::getWSSuffixes() const { return m_suffix; }
+const QStringList &WorkspaceMultiSelector::getWSSuffixes() const { return m_suffix; }
 
 void WorkspaceMultiSelector::setWSSuffixes(const QStringList &suffix) {
-  if (suffix != m_suffix) {
+  if (suffix != m_suffix)
     m_suffix = suffix;
```

Comment:
I just spotted that you could just do:
```suggestion
  m_suffix = suffix;
```
since it's a simple assignment so it's as quick as checking the if statement anyway.
But it is a pedantic change so it's good either way